docs/reference/contributing/target/target.md

```diff
@@ -34,9 +70,173 @@ Add the target description to `mbed-os\targets\targets.json` using keys that the
+    #define ISR_STACK_SIZE          (0x1000)
+    #endif
+
+    #ifndef OS_TASKCNT
+    #define OS_TASKCNT              14
+    #endif
+    #ifndef OS_MAINSTKSIZE
+    #define OS_MAINSTKSIZE          256
+    #endif
+    #ifndef OS_CLOCK
+    #define OS_CLOCK                120000000
+    #endif   
+```
+
+Please refer to your chosen MCU's reference manual for these values.
+
+### Startup and HAL
+
+Ensure that the drivers are present at the correct level in the directory structure:
+
+1. Add startup code and CMSIS specific headers. You may obtain the startup code and other CMSIS specific headers from the device manufacturer or the CMSIS packs from KEIL&reg;.
+1. Add peripherals to the `device_has_add` key in `targets.json` and include all relevant drivers for all these peripherals.
+1. Implement Mbed OS HAL APIs for all hardware peripherals mentioned in `device_has`.
+1. Ensure that `Objects.h` declares peripherals and is available in the `/api` directory.
+
+### Linker files
+
+Each supported toolchain listed in your new target's `supported_toolchains` configuration needs a linker file. Each compiler supports a different style of linker files with a different extension:
+
+- Add a scatter file, with the extension `.sct`, for Arm Compiler 5.
+- Add a linker script, with the extension `.ld`, for GCC Arm Embedded.
+- Add an IAR Linker file, with the extension `.icf`, for IAR EWARM.
+
+### Pin names
+
+Add peripheral names and pin names, following standard naming conventions for common peripherals. Define these pins in the `PinNames.h` header file at the board level. Please refer to the [sample directory structure](sample-directory-structure) for the directory structure.
+
+Define a default for each peripheral. Define the DAPLink serial pins as `STDIO_UART_RX` and `STDIO_UART_TX`. For example:
+
+```C
+#ifndef MBED_PINNAMES_H
+#define MBED_PINNAMES_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+    PA_0  = 0x00,
+    PA_1  = 0x01,
+    PA_1  = 0x01,
+<snip>
+
+    // Generic signals namings
+    LED1        = PB_2,
+    LED2        = PB_10,
+    LED_RED     = LED1,
+    LED_BLUE    = LED2,
+    USER_BUTTON = PC_13,
+
+    // Standardized button names
+    BUTTON1 = USER_BUTTON,
+
+    SERIAL_TX   = PA_8,
+    SERIAL_RX   = PA_12,
+
+    I2C_SCL     = PA_17,
+    I2C_SDA     = PA_18,
+
+    SPI_MOSI    = PA_4,
+    SPI_MISO    = PA_7,
+    SPI_SCK     = PA_6,
+    SPI_CS      = PA_16,
+
+    //DAPLink
+    STDIO_UART_RX = SERIAL_RX
+    STDIO_UART_TX = SERIAL_TX
+    USBRX      = STDIO_UART_RX,
+    USBTX      = STDIO_UART_TX,
+    SWDIO      = PA_26,
+    SWCLK      = PA_25,
+    NTRST      = PA_13,
+} PinName;
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
+```
+
+### Compiling with a supported compiler
+
+You can verify that the new board port compiles by using an [example application](/docs/development/tutorials/mbed-os-quick-start.html), and checking out the branch containing your port in the `mbed-os` subdirectory.
+
```

Comment:
@AnotherButler : Can you please add: "With the Mbed CLI, the command to use in the example's root directory is: mbed compile -t <TOOLCHAIN> -m <YOUR_NEW_TARGET's_NAME> -c . And if you face any build errors at this point, please add any missing headers, add/remove peripherals as required, ensure the linker scripts & startup code are in place. For simplicity, you can start with only the serial peripheral and add other peripherals once the Mbed OS port builds successfully."

Reply:
@AnotherButler, @ashok-rao  Don't add the `-c` it's not needed.